File: app/src/main/java/com/android/cy/pullrecyclerview/HomeActivity.java
package com.android.cy.pullrecyclerview;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.FragmentActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.android.cy.pullrecyclerview.adapter.HomeAdapters;
import com.android.cy.pullrecyclerview.view.CYStickyNavLayouts;


public class HomeActivity extends FragmentActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        CYStickyNavLayouts layout = (CYStickyNavLayouts) findViewById(R.id.head_home_layout);
        layout.setOnStartActivity(new CYStickyNavLayouts.OnStartActivityListener() {
            @Override
            public void onStart() {
                Intent intent = new Intent(HomeActivity.this, MainActivity.class);
                startActivity(intent);
            }
        });

        RecyclerView mHeadRecyclerView = (RecyclerView) findViewById(R.id.head_home_recyclerview);
        LinearLayoutManager layoutManager2 = new LinearLayoutManager(this);
        layoutManager2.setOrientation(LinearLayoutManager.HORIZONTAL);
        mHeadRecyclerView.setLayoutManager(layoutManager2);
        HomeAdapters mHomeAdapter = new HomeAdapters();
        mHeadRecyclerView.setAdapter(mHomeAdapter);
    }
}

File: app/src/main/java/com/android/cy/pullrecyclerview/HomeActivity.kt
package com.android.cy.pullrecyclerview

import android.content.Intent
import android.os.Bundle
import android.support.v4.app.FragmentActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView

import com.android.cy.pullrecyclerview.adapter.HomeAdapters
import com.android.cy.pullrecyclerview.view.CYStickyNavLayouts


class HomeActivity : FragmentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        val layout = findViewById(R.id.head_home_layout) as CYStickyNavLayouts
        layout.setOnStartActivity {
            val intent = Intent(this@HomeActivity, MainActivity::class.java)
            startActivity(intent)
        }

        val mHeadRecyclerView = findViewById(R.id.head_home_recyclerview) as RecyclerView
        val layoutManager2 = LinearLayoutManager(this)
        layoutManager2.orientation = LinearLayoutManager.HORIZONTAL
        mHeadRecyclerView.layoutManager = layoutManager2
        val mHomeAdapter = HomeAdapters()
        mHeadRecyclerView.adapter = mHomeAdapter
    }
}
